### PR TITLE
Resolve Log Race 

### LINKED ISSUE #455: logging data races
Enabling the race detector shows there are a lot of races happening around the logging, and this is the cause:

```
// Write writes the data in p to standard out and the log rotator.
func (l logWriter) Write(p []byte) (n int, err error) {
        os.Stdout.Write(p)
        go func() {
                internalLog <- string(p)
        }()
        return logRotator.Write(p)
}
```

The byte slice passed to Write is being modified later on, because the slog package will recycle and reuse the byte slice to minimize allocations.  Asynchronously reading this slice in another goroutine spawned by Write is incorrect and unsafe.

### PR DESCRIPTION
**Pull request by devchoplife**
Ths PR resolves #455
This PR fixes the Log race issue and also resolves the disappeareance of logs on log page in #571

### CODE REVIEW

File: log.go [hunk 1]
```diff
@@ -21,13 +21,11 @@ import (
...
 type logWriter struct{}
 
-var internalLog = make(chan string, 10)
-
 // Write writes the data in p to standard out and the log rotator.
 func (l logWriter) Write(p []byte) (n int, err error) {
 	os.Stdout.Write(p)
 	go func() {
-		internalLog <- string(p)
+		//internalLog <- string(p)
```
Comment (Sirmorrison) [inline log.go]: if this is no longer needed, kindly remove the go routine.

File: ui/page/log_page.go [hunk 2]
```diff
@@ -67,12 +66,17 @@ func (pg *LogPage) copyLogEntries(gtx C) {
 	}()
 }
 
-func (pg *LogPage) watchLogs(internalLog chan string) {
-	for l := range internalLog {
-		entry := l[:len(l)-1]
+func (pg *LogPage) watchLogs() {
+	//TODO
+	//add function to get log directory
+	logPath := "/home/devchoplife/.godcr/logs/godcr.log"
```
Comment (Sirmorrison) [inline ui/page/log_page.go]: i am not sure i will be able to test with this path.

**Review by Sirmorrison — CHANGES_REQUESTED** [review]
This is the display if i open the log page.
image

no logs are available as the `/home/devchoplife/.godcr/logs/godcr.log` does not exist in my machine.

image

The log path should not the be hard coded.

File: log.go (hunk 1 above)
Comment (devchoplife) [reply log.go:28]: I added a //TODO there .
A function will be added later to get the log directory

File: ui/page/log_page.go (hunk 2 above)
Comment (devchoplife) [reply ui/page/log_page.go]: I added a //TODO there .
A function will be added later to get the log directory

Comment (Sirmorrison) [reply ui/page/log_page.go]: When is the function going to be added and by who? With out this, this PR is not ready for review as it cannot be tested for functionality. Marking it as work in progress till it is ready.

**Comment by Clariceco** [comment]
@devchoplife what is the status of this? When will this be completed so that we can merge it?

**Comment by devchoplife** [comment]
> @devchoplife what is the status of this? When will this be completed so that we can merge it?

Waiting on https://github.com/planetdecred/dcrlibwallet/pull/206. It includes a function that gets log file path.

**Comment by devchoplife** [comment]
Works fine 
@C-ollins

**Review by Sirmorrison — APPROVED** [review]

**Review by beansgum — APPROVED** [review]

**Pull request merged**